c-lightning/2021-08-23-developer-call.md

```diff
@@ -0,0 +1,195 @@
+
+Good luck with the ongoing release, always a stressful time.
+
+# Individual updates (cont)
+
+I have implemented the check functions for invoice and invoice request and I have connected to the Bootstrap API, I will probably publish the NPM, the latest version of it and put out a Tweet. Maybe we will get some feedback. In parallel I work on implementing the Noise NK protocol. I am looking over [BOLT 8](https://github.com/lightningnetwork/lightning-rfc/blob/master/08-transport.md), that would help me to communicate with the node through web socket.
+
+Aditya is from [Summer of BItcoin](https://summerofbitcoin.org/). He has been working on the [BOLT 12](https://bolt12.org/) Javascript implementation particularly, decoding that. He has got the thing where you can fetch an invoice using the Bootstrap BOLT 12 REST server. But the next step is to try to connect through a web socket to a real Lightning node and speak enough Noise NK to talk to it to get an invoice. That involves several layers of fun but the first step is to speak Noise NK over the web socket so that you can get the encrypted transport happening. Fascinatingly that gives you enough to drive [Commando](https://github.com/lightningd/plugins/pull/280) from Javascript which is kind of interesting. In theory then he could write the code to do the Commando custom packets, which is just a random odd high number of packets, and given a rune could actually drive the node from Javascript. The goal is to be able to do a native onion message that given an offer you can fetch the invoice and then you can deconstruct that. The BOLT12 npm, that is regularly updated and is under active development now. It has all the pieces to decode it and everything else. There is also a Python pip BOLT 12 package that tries to track the Javascript version that has similar capabilities. In that case we have most of the pieces we need because the Python pieces talk Noise NK. In that case I’ve got the easy part because Christian has done that work a while ago. For the moment it uses bootstrap.bolt12.org which basically provides a nice tunnel. It has a fetch API so you can give it an invoice request, tell it where you want it to go and it will do the onion message for you and get the reply, spit it back at you. That’s the easy level and he is working on the next level which is pretty cool. There has already been over 200 downloads of the npm. People are obviously interested in it which is kind of cool. So hopefully there will be some good feedback. It is a little rough. The cool thing about the npm and the Python code is that they are generated from the spec. Aditya, most of his work was writing the generator to take the spec CSV that comes out of the BOLT repository and turn it into Javascript. There are layers on top to drive it. It is easy if the spec changes.
```

Comment:
```suggestion
Aditya is from [Summer of Bitcoin](https://summerofbitcoin.org/). He has been working on the [BOLT 12](https://bolt12.org/) Javascript implementation particularly, decoding that. He has got the thing where you can fetch an invoice using the Bootstrap BOLT 12 REST server. But the next step is to try to connect through a web socket to a real Lightning node and speak enough Noise NK to talk to it to get an invoice. That involves several layers of fun but the first step is to speak Noise NK over the web socket so that you can get the encrypted transport happening. Fascinatingly that gives you enough to drive [Commando](https://github.com/lightningd/plugins/pull/280) from Javascript which is kind of interesting. In theory then he could write the code to do the Commando custom packets, which is just a random odd high number of packets, and given a rune could actually drive the node from Javascript. The goal is to be able to do a native onion message that given an offer you can fetch the invoice and then you can deconstruct that. The BOLT12 npm, that is regularly updated and is under active development now. It has all the pieces to decode it and everything else. There is also a Python pip BOLT 12 package that tries to track the Javascript version that has similar capabilities. In that case we have most of the pieces we need because the Python pieces talk Noise NK. In that case I’ve got the easy part because Christian has done that work a while ago. For the moment it uses bootstrap.bolt12.org which basically provides a nice tunnel. It has a fetch API so you can give it an invoice request, tell it where you want it to go and it will do the onion message for you and get the reply, spit it back at you. That’s the easy level and he is working on the next level which is pretty cool. There has already been over 200 downloads of the npm. People are obviously interested in it which is kind of cool. So hopefully there will be some good feedback. It is a little rough. The cool thing about the npm and the Python code is that they are generated from the spec. Aditya, most of his work was writing the generator to take the spec CSV that comes out of the BOLT repository and turn it into Javascript. There are layers on top to drive it. It is easy if the spec changes.
```